package.json

```diff
@@ -18,31 +19,40 @@
   "bugs": {
     "url": "https://github.com/mrjono1/joi-to-typescript/issues"
   },
-  "main": "dist/index.js",
-  "types": "dist/index.d.ts",
+  "main": "./dist/main/index.js",
+  "module": "./dist/module/index.js",
+  "typings": "./dist/types/index.d.ts",
+  "exports": {
+    "import": "./dist/module/index.js",
+    "require": "./dist/main/index.js"
+  },
   "scripts": {
-    "build": "tsc",
+    "build:esm": "tsc --module es2020 --outDir ./dist/module",
+    "build:cjs": "tsc --module commonjs --outDir ./dist/main",
+    "build:dts": "tsc --declaration --emitDeclarationOnly --outDir ./dist/types",
+    "build": "yarn build:esm && yarn build:cjs && yarn build:dts",
     "format": "prettier --write \"src/**/*.ts\"",
     "lint": "eslint 'src/**'",
-    "test": "jest --config jest.config.js",
+    "test": "jest --config jest.config.cjs",
     "coverage": "yarn test --coverage --silent",
     "pub": "tsc && yarn publish"
   },
-  "dependencies": {
-    "joi": "^17.4.0"
-  },
   "devDependencies": {
-    "@types/jest": "^27.0.2",
+    "@types/jest": "^27.4.0",
     "@types/node": "^14.14.41",
-    "@typescript-eslint/eslint-plugin": "^4.15.2",
-    "@typescript-eslint/parser": "^4.22.0",
-    "eslint": "^7.21.0",
+    "@typescript-eslint/eslint-plugin": "^5.10.1",
+    "@typescript-eslint/parser": "^5.10.1",
+    "eslint": "^8.7.0",
     "eslint-config-prettier": "^8.3.0",
     "eslint-plugin-prettier": "^4.0.0",
-    "jest": "^27.1.0",
-    "prettier": "^2.2.1",
-    "ts-jest": "^27.0.5",
-    "typescript": "4.5.4"
+    "jest": "^27.4.7",
+    "joi": "^17.4.0",

```

Comment:
I think we need to remove this line
```suggestion
```

Reply:
I have Joi 17.3.0 in another project and I'm getting Joi version conflicts 
`Error: Cannot mix different versions of joi schemas`
but when its removed it cannot find Joi, am I misunderstanding peer dependencies?